packages/host/app/services/billing-service.ts

```diff
@@ -51,7 +52,13 @@ export default class BillingService extends Service {
   }
 
   get customerPortalLink() {
-    return this.stripeLinks.value?.customerPortalLink;
+    if (!this.subscriptionData?.stripeCustomerEmail) {
+      return this.stripeLinks.value?.customerPortalLink?.url;
+    }
+    const encodedEmail = encodeURIComponent(
+      this.subscriptionData.stripeCustomerEmail,
+    );
+    return `${this.stripeLinks.value?.customerPortalLink?.url}?prefilled_email=${encodedEmail}`;
```

Comment:
Avoid optional chaining as it can be confusing to read and reason about, especially when repeated. I find it better if we deal with the uncertainty right at the beginning and then operate on things we know they have a value:

```
if (!this.stripeLinks.value || 
    !this.stripeLinks.value.customerPortalLink || 
    !this.stripeLinks.value.customerPortalLink.url) {
  return undefined; 
}

... rest of the code 
```